src/banman.cpp

```diff
@@ -217,3 +230,35 @@ void BanMan::SetBannedSetDirty(bool dirty)
     LOCK(m_cs_banned); //reuse m_banned lock for the m_is_dirty flag
     m_is_dirty = dirty;
 }
+
+unsigned int BanMan::ZeroMisbehavior(CNetAddr net_addr)
+{
+    CSubNet sub_net(net_addr);
+    return ZeroMisbehavior(sub_net);
+}
+
+unsigned int BanMan::ZeroMisbehavior(CSubNet sub_net)
```

Comment:
I think we can put the `ZeroMisbehavior()` methods on `CNode`. Written that way, we can remove the `friend` visibility exposure that it looks like we don't really need here.

Reply:
Good point.

Reply:
They will be static, because they are not exactly actions taken on a specific node object per se, but that is ok.

Reply:
After looking at this I think I disagree with you. I would rather leave them where they are and make them private in banman. They are only consumed by banman, and in fact putting them in CNode, we would have to make them public to be accessible by banman, or protected, which again requires the friendship.

Making them public to eliminate the friend requirement then makes zeroing out the misbehavior accessible by any class, where having them private to banman only allows banman access. It makes sense to expose CNode protected structures to banman, as that is a relatively narrow surface area, IMHO.

Reply:
I think you're right about that after all. I was thinking about it in terms of unit testing and keeping `BanMan` out of `CNode`'s internals, but it looks like it makes sense in this case. Let's leave it as is.

Reply:
Cool.